meson.build

```diff
@@ -2390,48 +2392,51 @@ executable('systemd-sulogin-shell',
            install_dir : rootlibexecdir)
 
 ############################################################
+if want_tests == 'false'
+        message('Not compiling because tests is set to false')
+else
+        foreach tuple : tests
+                sources = tuple[0]
+                link_with = tuple[1].length() > 0 ? tuple[1] : [libshared]
+                dependencies = tuple[2]
+                condition = tuple.length() >= 4 ? tuple[3] : ''
+                type = tuple.length() >= 5 ? tuple[4] : ''
+                defs = tuple.length() >= 6 ? tuple[5] : []
+                incs = tuple.length() >= 7 ? tuple[6] : includes
+                timeout = 30
+
+                name = sources[0].split('/')[-1].split('.')[0]
+                if type.startswith('timeout=')
+                        timeout = type.split('=')[1].to_int()
+                        type = ''
+                endif
 
-foreach tuple : tests
-        sources = tuple[0]
-        link_with = tuple[1].length() > 0 ? tuple[1] : [libshared]
-        dependencies = tuple[2]
-        condition = tuple.length() >= 4 ? tuple[3] : ''
-        type = tuple.length() >= 5 ? tuple[4] : ''
-        defs = tuple.length() >= 6 ? tuple[5] : []
-        incs = tuple.length() >= 7 ? tuple[6] : includes
-        timeout = 30
-
-        name = sources[0].split('/')[-1].split('.')[0]
-        if type.startswith('timeout=')
-                timeout = type.split('=')[1].to_int()
-                type = ''
-        endif
-
-        if condition == '' or conf.get(condition) == 1
```

Comment:
OK, I can see why compiling the tests on some very slow hardware can be annoying. But please make that case less conspicuous: instead of moving the whole loop under the conditional, please add another case here, where you check if tests are disabled:
```
   if "tests disabled"
          message(...)
   else if condition == '' or conf.get(condition) == 1
          ...
   else
         message(...)
```

On of the advantages is that this way the meson configuration is still checked for consistency.

Reply:
It's not a "very slow hardware" thing, it's a "this requires gcrypt as a dependency" thing. :) 

As for the conditional, can do!